File: web/src/core/feature/exchange/domain/caseuse/GetTodayExchangeCaseUse.ts
import type { CupExchange } from "../entity/CupExchange";
import type { ExchangeRepository } from "../repository/ExchangeRepository";

export class GetTodayExchangeCaseUse {
    constructor(private readonly repository: ExchangeRepository) {}

    async execute(): Promise<CupExchange> {
        return this.repository.getToday();
    }
}

File: app/src/main/java/com/elitec/alejotaller/feature/exchange/domain/caseUse/GetTodayExchangeCaseUse.kt
package com.elitec.alejotaller.feature.exchange.domain.caseUse

import com.elitec.alejotaller.feature.exchange.domain.entity.CupExchange
import com.elitec.alejotaller.feature.exchange.domain.repository.ExchangeRepository

class GetTodayExchangeCaseUse(
    private val repository: ExchangeRepository
) {
    suspend operator fun invoke(): Result<CupExchange> = runCatching {
        repository.getToday()
    }
}
